src/main/scala/com/microsoft/ml/spark/lightgbm/LightGBMBase.scala

```diff
@@ -77,19 +107,19 @@ trait LightGBMBase[TrainedModel <: Model[TrainedModel]] extends Estimator[Traine
     }
   }
 
-  protected def getTrainingCols(): ListBuffer[String] = {
-    val trainingCols = ListBuffer(getLabelCol, getFeaturesCol)
+  protected def getTrainingCols(): ListBuffer[(String, DataType)] = {
+    val trainingCols = ListBuffer[(String, DataType)]((getLabelCol, DoubleType), (getFeaturesCol, VectorType))
```

Comment:
No mutability, Use a list of options and flatten

Reply:
done!